cognite/client/_api/simulators/logs.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import TYPE_CHECKING, overload
+
+from cognite.client._api_client import APIClient
+from cognite.client.data_classes.simulators.logs import SimulatorLog, SimulatorLogList
+from cognite.client.utils._experimental import FeaturePreviewWarning
+from cognite.client.utils._identifier import IdentifierSequence
+
+if TYPE_CHECKING:
+    from cognite.client import ClientConfig, CogniteClient
+
+
+class SimulatorLogsAPI(APIClient):
+    _RESOURCE_PATH = "/simulators/logs"
+
+    def __init__(self, config: ClientConfig, api_version: str | None, cognite_client: CogniteClient) -> None:
+        super().__init__(config, api_version, cognite_client)
+        self._warning = FeaturePreviewWarning(
+            api_maturity="General Availability", sdk_maturity="alpha", feature_name="Simulators"
+        )
+        self._RETRIEVE_LIMIT = 1
+
+    @overload
+    def retrieve(self, id: int) -> SimulatorLog | None: ...
+
+    @overload
+    def retrieve(
+        self,
+        id: int | Sequence[int],
+    ) -> SimulatorLogList | SimulatorLog | None: ...
```

Comment:
The second overload should reflect the return type when a sequence of ids is passed.
```suggestion
    @overload
    def retrieve(self, id: Sequence[int]) -> SimulatorLogList: ...
```

Reply:
What does it do? Could you provide an example?

The overload you have here is not invalid, it's just less useful than it can be.

Reply:
Agreed, it’s not incorrect—just not ideal. The plugin is quite dumb; it simply enforces a rule that prevents users from having overlapping parameter types across overloads. 

(Btw, this is just a suggestion, it shouldnt block you from fixing the comments on this PR, Luis 😄  ) 

Example: 
![carbon](https://github.com/user-attachments/assets/618fc64a-1acb-43d6-ab5e-253882279309)

Reply:
This seems quite useful. Could you create a separate PR adding this?